File: src/main/java/me/veritaris/delayer/Events/PlayerJoin.java
package me.veritaris.delayer.Events;

import me.veritaris.delayer.Delayer;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;

public class PlayerJoin implements Listener {

    @EventHandler(priority = EventPriority.LOWEST)
    public void onPlayerJoin(PlayerJoinEvent event) {
        Player eventPlayer = event.getPlayer();
        String player = String.format("%s", eventPlayer.getName());

        if (Delayer.getCommandsQueue().containsKey(player)) {
            Delayer.logger.info(String.format(
                    "Player %s joining", eventPlayer.getName()
            ));
            String command = Delayer.getCommandsQueue().get(player);
            Bukkit.getServer().dispatchCommand(Bukkit.getConsoleSender(), command);
            Delayer.logger.info(String.format(
                    "Issued command %s", command
            ));
            Delayer.getInstance().removeFromCommandQueue(player);
        }
    }
}

File: src/main/kotlin/me/veritaris/delayer/events/PlayerJoin.kt
package me.veritaris.delayer.events

import me.veritaris.delayer.Delayer.Companion.logger
import me.veritaris.delayer.Delayer.Companion.queueStorage
import org.bukkit.Bukkit
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent

class PlayerJoin : Listener {
    @EventHandler(priority = EventPriority.HIGHEST)
    fun onPlayerJoin(event: PlayerJoinEvent) {
        val eventPlayer = event.player
        val playerName = eventPlayer.name
        val delayedCommand = queueStorage.getCommandForPlayer(playerName) ?: return

        logger.info("Player ${eventPlayer.name} joining")
        Bukkit.getServer().dispatchCommand(Bukkit.getConsoleSender(), delayedCommand.command)
        logger.info("Issued command \"${delayedCommand.command}\" for player \"${eventPlayer.name}\"")
        queueStorage.removeDelayedCommandForPlayer(playerName)
    }
}